File: src-tauri/src/db/pool.rs
use sqlx::sqlite::{SqlitePool, SqlitePoolOptions};
use std::time::Duration;

/// 数据库连接池包装器
#[derive(Clone)]
pub struct DbPool {
    pool: SqlitePool,
}

impl DbPool {
    /// 创建新的数据库连接池
    pub async fn new(db_path: &str) -> Result<Self, sqlx::Error> {
        // 使用 file: URI 格式，并添加 mode=rwc 参数（read-write-create）
        // 这确保 SQLite 可以创建新文件
        let uri = if db_path.starts_with("file:") {
            db_path.to_string()
        } else {
            // ?mode=rwc 允许读取、写入和创建数据库文件
            // ?cache=shared 允许多个连接共享缓存
            format!("file:{}?mode=rwc&cache=shared", db_path)
        };

        tracing::info!("Creating SQLite pool with URI: {}", uri);

        let pool = SqlitePoolOptions::new()
            .max_connections(5)
            .min_connections(1)
            .acquire_timeout(Duration::from_secs(30))
            .idle_timeout(Duration::from_secs(600))
            .connect(&uri)
            .await
            .map_err(|e| {
                tracing::warn!("Failed to connect to SQLite: {}", e);
                e
            })?;

        tracing::info!("SQLite pool created successfully");
        Ok(Self { pool })
    }

    /// 获取底层连接池
    pub fn inner(&self) -> &SqlitePool {
        &self.pool
    }

}

File: backend/localdb/pool.go
package localdb

import (
	"database/sql"
	"fmt"
	"time"

	_ "modernc.org/sqlite"
)

// Pool SQLite 连接池包装器
type Pool struct {
	db *sql.DB
}

// NewPool 创建新的 SQLite 连接池
func NewPool(dbPath string) (*Pool, error) {
	uri := fmt.Sprintf("file:%s?mode=rwc&cache=shared", dbPath)

	db, err := sql.Open("sqlite", uri)
	if err != nil {
		return nil, fmt.Errorf("failed to open sqlite: %w", err)
	}

	db.SetMaxOpenConns(5)
	db.SetMaxIdleConns(1)
	db.SetConnMaxLifetime(10 * time.Minute)

	if err := db.Ping(); err != nil {
		return nil, fmt.Errorf("failed to ping sqlite: %w", err)
	}

	return &Pool{db: db}, nil
}

// Close 关闭连接池
func (p *Pool) Close() error {
	return p.db.Close()
}

// DB 获取底层数据库连接
func (p *Pool) DB() *sql.DB {
	return p.db
}
